packages/@coorpacademy-components/src/atom/icon/index.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {fas} from '@fortawesome/pro-solid-svg-icons';
+import {library} from '@fortawesome/fontawesome-svg-core';
+import toLower from 'lodash/fp/toLower';
+import merge from 'lodash/fp/merge';
+import getOr from 'lodash/fp/getOr';
+
+import Color from 'color';
+import style from './style.css';
+
+// FontAwesome documentation for where to find icon names: https://docs.fontawesome.com/web/add-icons/svg-icon-names
+// Search for FontAwesome icons: https://fontawesome.com/search
+
+library.add(fas);
+
+const DEFAULT_PRESET = 'm';
+const DEFAULT_WRAPPER_SIZE = 40;
+
+const SIZE_CONFIGS = {
+  s: {
+    faSize: 'sm',
+    wrapperSize: 32
+  },
+  m: {
+    faSize: 'lg',
+    wrapperSize: DEFAULT_WRAPPER_SIZE
+  },
+  xl: {
+    faSize: 'xl',
+    wrapperSize: 48
+  }
+};
+const hslObjectToHslString = ({h, s, l}) => `hsl(${h}, ${s}%, ${l}%)`;
+
+const getIconColor = (backgroundColor, defaultColor = 'hsl(0, 0%, 32%)') => {
+  if (!backgroundColor) return defaultColor;
+
+  try {
+    const color = Color(backgroundColor);
+    const adjustedColor = color.lightness(32);
+    return hslObjectToHslString(adjustedColor.hsl());
+  } catch (error) {
```

Comment:
c'est quoi les errors qu'on risque?

Reply:
Par exemple, si le dev s’amuse à mettre une valeur de couleur qui n’a aucun sens (d’ailleurs j’ai mis un fixture qui montre ca) au lieu d’avoir une erreur (Error: Unable to parse color from string "#D9F4Fsdfsss7 ») on aura une valeur par défaut

Reply:
justement on veut l'erreur, dans les tests, et que ça soit corrigé. Avec ce comportement l'erreur est quasi invisible